test/cri-containerd/container_virtual_device_test.go

```diff
@@ -0,0 +1,327 @@
+	}
+}
+
+// verifyGPUIsNotPresent is a helper function that runs a command in the container
+// to verify that there are no GPUs present in the container and fails the running test
+// if any are found
+func verifyGPUIsNotPresent(t *testing.T, client runtime.RuntimeServiceClient, ctx context.Context, containerID string) {
+	execReq := makeGPUExecCommand(containerID)
+	response := execSync(t, client, ctx, execReq)
+	if len(response.Stderr) == 0 {
+		t.Fatal("expected to see an error as file /dev/nvidia0 should not exist, instead saw none")
+	} else if len(response.Stdout) != 0 {
+		t.Fatal("expected to not see GPU device on container, but some are present")
+	}
+}
+
+// findTestDevices returns the first nvidia pcip device on the host
+func findTestNvidiaGPUDevice() (string, error) {
+	out, err := exec.Command(
+		"powershell",
+		`(Get-PnpDevice -presentOnly | where-object {$_.InstanceID -Match 'PCIP\\VEN_10DE.*'})[0].InstanceId`,
+	).Output()
+	if err != nil {
+		return "", err
+	}
+	return strings.TrimSpace(string(out)), nil
+}
+
+func getGPUPodRequestLCOW(t *testing.T) *runtime.RunPodSandboxRequest {
+	return &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Namespace: testNamespace,
+			},
+			Annotations: map[string]string{
+				"io.microsoft.virtualmachine.lcow.kerneldirectboot":                  "false",
+				"io.microsoft.virtualmachine.computetopology.memory.allowovercommit": "false",
+				"io.microsoft.virtualmachine.lcow.preferredrootfstype":               "initrd",
+				"io.microsoft.virtualmachine.devices.virtualpmem.maximumcount":       "0",
+				"io.microsoft.virtualmachine.lcow.vpcienabled":                       "true",
+				// we believe this is a sufficiently large high MMIO space amount for this test.
+				// if a given gpu device needs more, this test will fail to create the container
+				// and may hang.
+				"io.microsoft.virtualmachine.computetopology.memory.highmmiogapinmb": "64000",
+				"io.microsoft.virtualmachine.lcow.bootfilesrootpath":                 testGPUBootFiles,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+}
+
+func getGPUContainerRequest(t *testing.T, podID string, podConfig *runtime.PodSandboxConfig, device *runtime.Device) *runtime.CreateContainerRequest {
+	return &runtime.CreateContainerRequest{
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Devices: []*runtime.Device{
+				device,
+			},
+			Linux: &runtime.LinuxContainerConfig{},
+			Annotations: map[string]string{
+				"io.microsoft.container.gpu.capabilities": "utility",
+			},
+		},
+		PodSandboxId:  podID,
+		SandboxConfig: podConfig,
+	}
+}
+
+func Test_RunContainer_VirtualDevice_GPU_LCOW(t *testing.T) {
+	requireFeatures(t, featureLCOW)
```

Comment:
`requireFeatures` takes a vararg of features, so you don't need to call it twice.

Reply:
Fixed